src/plugins/kibana/server/routes/api/index_pattern_endpoints.js

```diff
@@ -0,0 +1,258 @@
+const esErrors = require('elasticsearch').errors;
+const Boom = require('boom');
+const Joi = require('joi');
+const _ = require('lodash');
+const Promise = require('bluebird');
+const getMappings = require('../../lib/get_mappings');
+const stitchPatternAndMappings = require('../../lib/stitch_pattern_and_mappings');
+const {templateToPattern, patternToTemplate} = require('../../lib/convert_pattern_and_template_name');
+const removeDeprecatedFieldProps = require('../../lib/remove_deprecated_field_props');
+const indexPatternSchema = require('../../lib/schemas/index_pattern_schema');
+
+export default function (server) {
+
+  let handleESError = function (error) {
+    if (error instanceof esErrors.ConnectionFault ||
+      error instanceof esErrors.ServiceUnavailable ||
+      error instanceof esErrors.NoConnections ||
+      error instanceof esErrors.RequestTimeout) {
+      return Boom.serverTimeout(error);
+    } else if (error instanceof esErrors.Conflict || _.contains(error.message, 'index_template_already_exists')) {
+      return Boom.conflict(error);
+    } else if (error instanceof esErrors[403]) {
+      return Boom.forbidden(error);
+    } else if (error instanceof esErrors.NotFound) {
+      return Boom.notFound(error);
+    } else if (error instanceof esErrors.BadRequest || error instanceof TypeError) {
+      return Boom.badRequest(error);
+    } else {
+      return error;
+    }
+  };
+
```

Comment:
Can you break these routes out into their own files, under an index_pattern directory? Then just include them in this file?